### PR TITLE
#158 create logic for view ca

### PR DESCRIPTION
**Pull request by pinkevmladchy**

### CODE REVIEW

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 1]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+        transportId: selectedRecord.sideOfAccident.transportId,
+        driverLicenseSerial: selectedRecord.sideOfAccident.driverLicenseSerial,
+      },
+      accidentCircumstances: selectedRecord.accidentCircumstances,
+      trafficRuleId: selectedRecord.trafficRuleId,
+      driverExplanation: selectedRecord.driverExplanation,
+      witnesses: {
+        lastName: selectedRecord.witnesses[0].lastName,
+        firstName: selectedRecord.witnesses[0].firstName,
+        phoneNumber: selectedRecord.witnesses[0].phoneNumber,
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: Create let witness and use witness.firstname...

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 2]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+        street: selectedRecord.address.street,
+        crossStreet: selectedRecord.address.crossStreet,
+        coordinatesOfLatitude: selectedRecord.address.coordinatesOfLatitude,
+        coordinatesOfLongitude: selectedRecord.address.coordinatesOfLongitude,
+        isInCity: selectedRecord.address.isInCity,
+        isIntersection: selectedRecord.address.isIntersection,
+      },
+      sideOfAccident: {
+        email: selectedRecord.sideOfAccident.email,
+        transportId: selectedRecord.sideOfAccident.transportId,
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: Create let sideOfAccident and use sideOfAccident.email...

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 3]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+  async populateForm(selectedRecord: CarAccident) {
+    this.DataForm.setValue({
+      serialNumber: selectedRecord.serialNumber,
+      inspectorId: selectedRecord.inspectorId,
+      registrationDateTime: selectedRecord.registrationDateTime,
+      address: {
+        city: selectedRecord.address.city,
+        district: selectedRecord.address.district,
+        street: selectedRecord.address.street,
+        crossStreet: selectedRecord.address.crossStreet,
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: Create let address and use address.city..

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 4]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+  @Input() set setAdd(isAdd: boolean) {
+    this.isAdd = isAdd;
+  }
+
+  @Input() set setData(protocol: CarAccident){
+      this.protocolCA = protocol;
+      if(this.isAdd === false){
+        if(this.protocolCA){
+          this.populateForm(protocol);
+          console.log(this.transportForm.value);
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: delete

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 5]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
   }
 
+  @Input() set setAdd(isAdd: boolean) {
+    this.isAdd = isAdd;
+  }
+
+  @Input() set setData(protocol: CarAccident){
+      this.protocolCA = protocol;
+      if(this.isAdd === false){
+        if(this.protocolCA){
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: use condition isAdd&&protocolCA

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 6]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 })
 export class ViewCAComponent implements OnInit {
 
-  isVisible = false;
-  constructor() { }
+  public isVisible = false;
+  isAdd = false;
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: add public

File: src/app/components/inspector/view-ca-modal/view-ca.component.ts [hunk 7]
```diff
@@ -7,21 +11,136 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+  protocolCA: CarAccident;
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/view-ca-modal/view-ca.component.ts]: add public

File: src/app/components/inspector/inspector-list-of-car-accidents/inspector-list-of-car-accidents.component.ts [hunk 8]
```diff
@@ -9,30 +13,54 @@ import { InspectorService } from 'src/app/services/inspector.service';
...
       data => {
         if (data.length == 0) {
-          this.isAccidentListEmpty = true;
+          this.isAccidentListEmpty = true;                    
         }
         else {
           this.accidentList = data;
         }
       },
       error => { }
```
Comment (SashaKorniichuk) [inline src/app/components/inspector/inspector-list-of-car-accidents/inspector-list-of-car-accidents.component.ts:53]: add toastr if list empty using error.err.message

**Review by SashaKorniichuk — CHANGES_REQUESTED** [review]

**Review by SashaKorniichuk — APPROVED** [review]

**Review by ICHTLAY — APPROVED** [review]

**Pull request merged**